src/zh/komiic/src/eu/kanade/tachiyomi/extension/zh/komiic/Komiic.kt

```diff
@@ -0,0 +1,291 @@
+     */
+    private fun comicByIDRequest(id: String): Request {
+        val payload = Payload(
+            operationName = "comicById",
+            variables = ComicByIdVariables(id),
+            query = QUERY_COMIC_BY_ID,
+        ).toJsonRequestBody()
+        return POST(queryAPIUrl, headers, payload)
+    }
+
+    /**
+     * 根據 ID 解析搜索來的漫畫
+     * Parse the comic based on the ID.
+     */
+    private fun parseComicByID(response: Response): MangasPage {
+        val res = response.parseAs<Data<ComicByIDResponse>>()
+        val entries = mutableListOf<SManga>()
+        val comic = res.data.comic.toSManga()
+        entries.add(comic)
+        val hasNextPage = entries.size == pageSize
+        return MangasPage(entries, hasNextPage)
+    }
+
+    // Search
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val payload = Payload(
+            operationName = "searchComicAndAuthorQuery",
+            variables = SearchVariables(query),
+            query = QUERY_SEARCH,
+        ).toJsonRequestBody()
+        return POST(queryAPIUrl, headers, payload)
+    }
+
+    override fun fetchSearchManga(
+        page: Int,
+        query: String,
+        filters: FilterList,
+    ): Observable<MangasPage> {
+        return if (query.startsWith(PREFIX_ID_SEARCH)) {
+            val mangaId = query.substringAfter(PREFIX_ID_SEARCH)
+            client.newCall(comicByIDRequest(mangaId))
+                .asObservableSuccess()
+                .map(::parseComicByID)
+        } else {
+            super.fetchSearchManga(page, query, filters)
+        }
+    }
+
+    override fun searchMangaParse(response: Response) = parseSearchList<SearchResponse>(response)
+
+    // Comic details
+    override fun mangaDetailsRequest(manga: SManga) = comicByIDRequest(manga.url.substringAfterLast("/"))
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val res = response.parseAs<Data<ComicByIDResponse>>()
+        val comic = res.data.comic.toSManga()
+        return comic
+    }
+
+    override fun getMangaUrl(manga: SManga) = "$baseUrl${manga.url}"
+
+    /**
+     * 解析日期
+     * Parse date
+     */
+    private fun parseDate(dateStr: String): Long {
+        val sdf = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.US)
+        sdf.timeZone = TimeZone.getTimeZone("UTC")
+        return sdf.parse(dateStr)?.time ?: 0L
+    }
+
+    /**
+     * 解析章節列表
+     * Parse chapter list
+     */
+    private inline fun <reified T : ChaptersResult> parseChapterList(response: Response, comicUrl: String): List<SChapter> {
+        val res = response.parseAs<Data<T>>()
+        val comics = res.data.chapters
+
+        val tChapters = comics.filter { it.type == "chapter" }
+        val tBooks = comics.filter { it.type == "book" }
+
+        val entries = (tChapters + tBooks).map { chapter ->
+            SChapter.create().apply {
+                url = "$comicUrl/chapter/${chapter.id}/page/1"
```

Comment:
consider only saving id, otherwise the relative url (also would need to override getChapterUrl for webview url)

Reply:
the comicUrl is already relative url
![image](https://github.com/user-attachments/assets/ab4fbd68-88ff-4224-ac0c-7c8f613edb03)